tools/lib/ensure-coreutils.sh

```diff
@@ -4,17 +4,34 @@
 # Ensures GNU coreutils are available in PATH.
 # May add to PATH, or exit with a message to stderr.
 #
-# We don't yet have good docs for handling failure here if on
-# Windows or macOS, so until then this isn't suitable for scripts
-# used in normal development -- only for maintainer-facing scripts.
+# The GNU coreutils are a basic piece of infrastructure for having
+# a reasonable 21st-century shell scripting environment, so we
+# freely invoke this in any of our scripts that need `readlink -f`
+# or other features not always found without them.
+#
+# On any GNU/Linux system this is of course a non-issue, and nor
+# is it on Windows: we inevitably require Git, and Git for Windows
+# comes with a GNU environment called "Git BASH", based on MSYS2.
```

Comment:
`s/and nor/nor/;`

Reply:
Hmm, somehow that version feels ungrammatical to me! Yet it's true "and nor" is not common.

I think it's related to the fact the first clause isn't really in the negative. If this version bugs you, I can reword.

Reply:
(done)